cpp/src/arrow/compute/kernels/scalar_cast_numeric.cc

```diff
@@ -641,6 +684,12 @@ std::shared_ptr<CastFunction> GetCastToDecimal128() {
   DCHECK_OK(func->AddKernel(Type::DOUBLE, {float64()}, sig_out_ty,
                             CastFunctor<Decimal128Type, DoubleType>::Exec));
 
+  // Cast from integer
+  for (const std::shared_ptr<DataType>& in_ty : IntTypes()) {
+    auto exec = GenerateInteger<CastFunctor, Decimal128Type>(in_ty->id());
+    DCHECK_OK(func->AddKernel(in_ty->id(), {in_ty}, sig_out_ty, std::move(exec)));
+  }
```

Comment:
No need to `std::move(exec)` as it is already an rvalue (function pointer).

Reply:
`exec` is an std::function object with 32 bytes in size. Not a real issue, I prefer keeping the move operation.